core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/pythonUDF/PythonUDFOpExec.java

```diff
@@ -15,218 +15,70 @@
-            }
-
-            int portNumber = getFreeLocalPort();
-            Location location = new Location(URI.create("grpc+tcp://localhost:" + portNumber));
-            List<String> args = new ArrayList<>(
-                    Arrays.asList(PYTHON, DAEMON_SCRIPT_PATH, Integer.toString(portNumber), pythonScriptPath)
-            );
-
-            ProcessBuilder processBuilder = new ProcessBuilder(args).inheritIO();
-            // Start Flight server (Python process)
-            processBuilder.start();
-            // Connect to server
-            boolean connected = false;
-            int tryCount = 0;
-            while (!connected && tryCount < MAX_TRY_COUNT) {
-                try {
-                    Thread.sleep(WAIT_TIME_MS);
-                    flightClient = FlightClient.builder(globalRootAllocator, location).build();
-                    String message = new String(
-                            flightClient.doAction(new Action("healthcheck")).next().getBody(), StandardCharsets.UTF_8);
-                    connected = message.equals("Flight Server is up and running!");
-                } catch (Exception e) {
-                    System.out.println("Flight Client:\tNot connected to the server in this try.");
-                    flightClient.close();
-                    tryCount++;
-                }
-            }
-            if (tryCount == MAX_TRY_COUNT)
-                throw new RuntimeException("Exceeded try limit of " + MAX_TRY_COUNT +" when connecting to Flight Server!");
-        } catch (Exception e) {
-            e.printStackTrace();
-            try {
-                if (isDynamic) deleteTempFile(pythonScriptPath);
-            } catch (Exception innerException) {
-                innerException.printStackTrace();
-            }
-        }
-
-        // Send user args to Server.
-        List<String> userArgs = new ArrayList<>();
-        if (inputColumns != null) userArgs.addAll(inputColumns);
-        if (outputColumns != null) {
-            for (Attribute a : outputColumns) userArgs.add(a.getName());
-        }
-        if (outerFilePaths != null) userArgs.addAll(outerFilePaths);
-
-        Schema argsSchema = new Schema(Collections.singletonList(new Attribute("args", AttributeType.STRING)));
-        Queue<Tuple> argsTuples = new LinkedList<>();
-        for (String arg : userArgs) {
-            argsTuples.add(new Tuple(argsSchema, Collections.singletonList(arg)));
-        }
-
         try {
-            writeArrowStream(flightClient, argsTuples, globalRootAllocator, convertAmber2ArrowSchema(argsSchema), "args", batchSize);
-            flightClient.doAction(new Action("open")).next().getBody();
-        } catch (Exception e) {
+            PYTHON = Files.walk(Utils.amberHomePath().resolve("../"), 5)
+                    .filter((path) -> path.getFileName().toString().equals("python3")).map(Path::toString)
+                    .findFirst().orElse("python3");
```

Comment:
I don't think this is a good solution... even for a shor term hacky solution... maybe we should just install texera in an environment where python command is available, or maybe we should create a config file and put the python interpreter path there

Reply:
changed it to read from `core/config/web.conf`.